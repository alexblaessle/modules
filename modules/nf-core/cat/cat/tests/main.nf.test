nextflow_process {

    name "Test Process CAT_CAT"
    script "modules/nf-core/cat/cat/main.nf"
    process "CAT_CAT"
    tag "module"
    tag "modules_nfcore"
    tag "cat_cat"

    test("test_cat_unzipped_unzipped") {
        when {
            params {
                outdir   = "${outputDir}"
            }
            process {
                """
                input[0] =
                    [
                        [ id:'test', single_end:true ],
                        [
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['genome_sizes'], checkIfExists: true)
                        ]
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("test_cat_zipped_zipped") {
        when {
            params {
                outdir   = "${outputDir}"
            }
            process {
                """
                input[0] =
                    [
                        [ id:'test', single_end:true ],
                        [
                            file(params.test_data['sarscov2']['genome']['genome_gff3_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['contigs_genome_maf_gz'], checkIfExists: true)
                        ]
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_zipped_unzipped") {
        config './nextflow_zipped_unzipped.config'

        when {
            params {
                outdir   = "${outputDir}"
            }
            process {
                """
                input[0] =
                    [
                        [ id:'test', single_end:true ],
                        [
                            file(params.test_data['sarscov2']['genome']['genome_gff3_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['contigs_genome_maf_gz'], checkIfExists: true)
                        ]
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_cat_unzipped_zipped") {
        config './nextflow_unzipped_zipped.config'
        when {
            params {
                outdir   = "${outputDir}"
            }
            process {
                """
                input[0] =
                    [
                        [ id:'test', single_end:true ],
                        [
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['genome_sizes'], checkIfExists: true)
                        ]
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test_cat_one_file_unzipped_zipped") {
        config './nextflow_unzipped_zipped.config'
        when {
            params {
                outdir   = "${outputDir}"
            }
            process {
                """
                input[0] =
                    [
                        [ id:'test', single_end:true ],
                        [
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                        ]
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}

