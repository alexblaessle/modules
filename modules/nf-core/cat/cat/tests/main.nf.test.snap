{
    "test_cat_unzipped_unzipped": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.fasta:md5,f44b33a0e441ad58b2d3700270e2dbe2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ],
                "file_out": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.fasta:md5,f44b33a0e441ad58b2d3700270e2dbe2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ]
            }
        ],
        "timestamp": "2023-10-16T12:58:15.77604818"
    },
    "test_cat_zipped_zipped": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.gz:md5,1c3e5c5ddf6ebcaf56f3283905269974"
                    ]
                ],
                "1": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ],
                "file_out": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.gz:md5,1c3e5c5ddf6ebcaf56f3283905269974"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ]
            }
        ],
        "timestamp": "2023-10-16T12:58:28.552619778"
    },
    "test_cat_zipped_unzipped": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "cat.txt:md5,c439d3b60e7bc03e8802a451a0d9a5d9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ],
                "file_out": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "cat.txt:md5,c439d3b60e7bc03e8802a451a0d9a5d9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ]
            }
        ],
        "timestamp": "2023-10-16T12:56:53.482831598"
    },
    "test_cat_one_file_unzipped_zipped": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "cat.txt.gz:md5,da9c36c3e304f1591029f88a2bf2d9a6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ],
                "file_out": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "cat.txt.gz:md5,da9c36c3e304f1591029f88a2bf2d9a6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ]
            }
        ],
        "timestamp": "2023-10-16T12:59:09.214997947"
    },
    "test_cat_unzipped_zipped": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "cat.txt.gz:md5,f807b0f5a14e52f00d3e26aa3ad74826"
                    ]
                ],
                "1": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ],
                "file_out": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "cat.txt.gz:md5,f807b0f5a14e52f00d3e26aa3ad74826"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,115ed6177ebcff24eb99d503fa5ef894"
                ]
            }
        ],
        "timestamp": "2023-10-16T12:58:57.166235206"
    }
}